File: adventofcode/java/year2016/day13/Day13Test.java
package adventofcode.java.year2016.day13;

import java.io.FileReader;
import java.io.IOException;

import org.junit.Test;
import org.junit.Assert;

public class Day13Test {
    @Test
    public void part1Actual() throws IOException {
        try (var input = new FileReader("adventofcode/data/year2016/day13/actual.in")) {
            var output = "92";
            var result = Day13.part1(input);
            Assert.assertEquals("no error", "", result.error);
            Assert.assertEquals("correct output", output, result.answer);
        }
    }

     @Test
     public void part2Actual() throws IOException {
         try (var input = new FileReader("adventofcode/data/year2016/day13/actual.in")) {
             var output = "124";
             var result = Day13.part2(input);
             Assert.assertEquals("no error", "", result.error);
             Assert.assertEquals("correct output", output, result.answer);
         }
     }
}

File: adventofcode/cpp/year2019/day13/day13_test.cc
#include "adventofcode/cpp/year2019/day13/day13.h"

#include <fstream>
#include <string>

#include "gtest/gtest.h"

#include "adventofcode/cpp/adventofcode.h"

TEST(Year2019Day13, Part1Actual) {
  std::ifstream input("adventofcode/data/year2019/day13/actual.in");
  std::string output = "216";
  adventofcode::answer_t a = day13::part1(input);
  EXPECT_EQ("", a.error);
  EXPECT_EQ(output, a.answer);
  input.close();
}

TEST(Year2019Day13, Part2Actual) {
  std::ifstream input("adventofcode/data/year2019/day13/actual.in");
  std::string output = "10025";
  adventofcode::answer_t a = day13::part2(input);
  EXPECT_EQ("", a.error);
  EXPECT_EQ(output, a.answer);
  input.close();
}
